forge-ai/src/main/java/forge/ai/ManaPaymentService.java

```diff
@@ -0,0 +1,1437 @@
+                                isHighOutput = true;
+                            }
+                        } catch (NumberFormatException e) {
+                            // Try to use AbilityUtils.calculateAmount as a fallback for complex expressions
+                            potentialManaOutput = AbilityUtils.calculateAmount(card, amountStr, m);
+                            if (potentialManaOutput > 1) {
+                                isHighOutput = true;
+                            }
+                        }
+                    }
+                }
+
+                AbilitySub sub = m.getSubAbility();
+                // We really shouldn't be hardcoding names here. ChkDrawback should just return true for them
+                if (sub != null && !card.getName().equals("Pristine Talisman") && !card.getName().equals("Zhur-Taa Druid")) {
+                    if (!SpellApiToAi.Converter.get(sub).chkDrawbackWithSubs(ai, sub)) {
+                        continue;
+                    }
+                    needsLimitedResources = true; // TODO: check for good drawbacks (gainLife)
+                }
+                usableManaAbilities++;
+            }
+
+            // Special case for Serra's Sanctum and similar cards
+            if (isHighOutput && potentialManaOutput > 0) {
+                // If this source can produce a lot of mana at once, prioritize it
+                highOutputManaSources.add(card);
+                // Store the potential output in our map
+                manaOutputPotential.put(card, potentialManaOutput);
+                continue;
+            }
+
+            if (unpreferredCost) {
+                useLastManaSources.add(card);
+            } else if (needsLimitedResources) {
+                otherManaSources.add(card);
+            } else if (producesAnyColor) {
+                anyColorManaSources.add(card);
+            } else if (usableManaAbilities == 1) {
+                if (manaAbilities.get(0).getManaPart().mana(manaAbilities.get(0)).equals("C")) {
+                    colorlessManaSources.add(card);
+                } else {
+                    oneManaSources.add(card);
+                }
+            } else if (usableManaAbilities == 2) {
+                twoManaSources.add(card);
+            } else if (usableManaAbilities == 3) {
+                threeManaSources.add(card);
+            } else if (usableManaAbilities == 4) {
+                fourManaSources.add(card);
+            } else {
+                fiveManaSources.add(card);
+            }
+        }
+
+        // Sort high output mana sources by potential output (highest first)
+        highOutputManaSources.sort((a, b) -> Integer.compare(
+            manaOutputPotential.getOrDefault(b, 0),
+            manaOutputPotential.getOrDefault(a, 0)
+        ));
+
+        // Add high output sources first - they're most efficient
+        sortedManaSources.addAll(sortedManaSources.size(), highOutputManaSources);
```

Comment:
does this lead to negative impacts?
I guess you added it to nudge it into passing your test...?

but does this mean instead AI will now waste _Serra's Sanctum_ in Main1 for a white one drop and then fail to pay for something bigger in M2? 👎